tools/metrics/main.go

```diff
@@ -17,54 +21,81 @@ import (
 	"code.cloudfoundry.org/lager"
 )
 
-func handler(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Cache-Control", "max-age=0,no-store,no-cache")
-	w.Header().Set("Content-Type", "application/json")
-	json.NewEncoder(w).Encode(struct {
-		OK bool
-	}{
-		OK: true,
-	})
+func initPrometheus() (*prometheus.Registry, http.Handler) {
+	registry := prometheus.NewRegistry()
+	handler := promhttp.HandlerFor(registry, promhttp.HandlerOpts{})
+	return registry, handler
 }
 
-func server() error {
-	addr := ":" + os.Getenv("PORT")
-	http.HandleFunc("/", handler)
-	return http.ListenAndServe(addr, nil)
+func getHTTPPort() int {
+	portStr := os.Getenv("PORT")
+	if portStr != "" {
+		port, err := strconv.Atoi(portStr)
+		if err != nil {
+			log.Fatalln("PORT is invalid")
+			return 0
+		}
+		return port
+	}
+
+	return 8080
+}
+
+func runHTTPServer(port int, metricsHandler http.Handler) {
+	addr := fmt.Sprintf(":%d", port)
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Cache-Control", "max-age=0,no-store,no-cache")
```

Comment:
Perhaps include `s-maxage=0` here, too, as we're behind CloudFront.

Reply:
Originally we weren't, did this change recently? Also this is a copy-paste from the original code and I don't think it matters as this is checked by gorouter internally as part of the http check and there is no cache in play. (Or I'm badly mistaken)

Reply:
Minor point - this won't be behind CloudFront. CloudFront is only used by cdn-routes it's not in place for *.cloudapps.digital.

Reply:
I'll leave this as is

Reply:
@alext Thanks - I'd made an assumption we had traffic parity between BYOD and non-BYOD apps.

Reply:
It's always dangerous to make assumptions, so don't make them :)